### PR TITLE
Register

### PR DESCRIPTION
**Pull request by Avetiq30**
https://www.notion.so/B-User-Registration-Api-fe8e5623dc9b40d0a8510174ca6654b7?pvs=4

### CODE REVIEW

File: src/user/user.controller.ts [hunk 1]
```diff
@@ -18,17 +21,21 @@ export class UserController {
   async registerUser(@Body() createUserDto: CreateUserDto) {
     return this.userService.registerUser(createUserDto);
   }
+  @Get()
+  @UseGuards(JwtAuthGuard)
+  async getAll(@Req() req) {
+    const role = req.user.role;
+    if (role !== 'admin') {
+      throw new Error('error');
+    }
```
Comment (VahagShahinyan) [inline src/user/user.controller.ts:30]: Add Gurad

File: test/user.e2e-spec.ts [hunk 2]
```diff
@@ -47,31 +47,34 @@ describe('UserController', () => {
...
+  //       .post('/user/register')
+  //       .send(loginData)
+  //       .expect(HttpStatus.BAD_REQUEST);
 
-      expect(response.body.message).toBe(USER_WITH_THIS_EMAIL);
-    });
-  });
+  //     expect(response.body.message).toBe(USER_WITH_THIS_EMAIL);
+  //   });
+  // });
```
Comment (VahagShahinyan) [inline test/user.e2e-spec.ts:79]: Why it is commented ?

File: src/auth/jwt.strategy.ts [hunk 3]
```diff
@@ -15,4 +15,8 @@ export class JwtStrategy extends PassportStrategy(Strategy) {
       secretOrKey: configService.get('JWT_SECRET'),
     });
   }
+  async validate(payload: any) {
```
Comment (VahagShahinyan) [inline src/auth/jwt.strategy.ts:18]: ```validate(payload: any) => validate(payload: {email:string, role: string})```

Comment (Avetiq30) [reply src/auth/jwt.strategy.ts:18]: ++

File: src/user/user.controller.ts (hunk 1 above)
Comment (Avetiq30) [reply src/user/user.controller.ts:30]: sran der chem hase esi uxaki pordzum ey

File: test/user.e2e-spec.ts (hunk 2 above)
Comment (Avetiq30) [reply test/user.e2e-spec.ts:79]: ++

**Pull request merged**